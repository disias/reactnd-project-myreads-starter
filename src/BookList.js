import React,{Component} from 'react'
import PropTypes from 'prop-types'
import Book from './Book'

/**
 *
 *
 * @export
 * @class BookList
 * @extends {Component}
 */
class BookList extends Component {

  //PropType  BookList
  static propTypes = {
    books: PropTypes.array.isRequired,
    shelves: PropTypes.array.isRequired,
    onMoveBook : PropTypes.func.isRequired
  };

  componentWillUnmount() {
    console.log('BookList-componentWillUnmount')
  }

  componentDidMount(){
    console.log('BookList-componentDidMount')
  }



   render(){
    //tribuição via desestruturação é uma expressão JavaScript que possibilita
    //extrair dados de arrays ou objetos em variáveis distintas.
    const { books , shelves , onMoveBook } = this.props;
    // Alimenta o booklist com books
    const elements = books.map( book =>(
      <li key={book.id}>
        <Book book={book} shelves={shelves} onMoveBook={onMoveBook} />
      </li>
    ))

    return (
      <div className="bookshelf-books">
        <ol className="books-grid">
          {elements}
        </ol>
      </div>
    )
  }
}


export default BookList;
