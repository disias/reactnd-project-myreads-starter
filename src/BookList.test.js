import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {Route,BrowserRouter} from 'react-router-dom';
import BookList from './BookList';

describe('ListBook', () => {

   it('renders booklist', () => {
       const output = renderer.create(
        <BrowserRouter>
          <Route path='/' render={()=>(
            <BookList books={book} shelves={shelves} onMoveBook={onMoveBook} /> )} />
        </BrowserRouter>
      )
       expect(output).toMatchSnapshot()
   })

})
