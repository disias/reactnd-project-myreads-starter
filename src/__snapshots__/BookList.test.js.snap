// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBook renders booklist 1`] = `
<div
  className="bookshelf-books"
>
  <ol
    className="books-grid"
  >
    <li>
      <div
        className="book"
      >
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            data-teste="imageLinks"
            style={
              Object {
                "backgroundImage": "url(http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api)",
                "height": 192,
                "width": 128,
              }
            }
          />
          <div
            className="book-shelf-changer"
          >
            <select
              defaultValue="currentlyReading"
              onChange={[Function]}
            >
              <option
                disabled={true}
              >
                Move to...
              </option>
              <option
                data-teste="shelf"
                value="currentlyReading"
              >
                Currently Reading
              </option>
              <option
                data-teste="shelf"
                value="wantToRead"
              >
                Want to Read
              </option>
              <option
                data-teste="shelf"
                value="read"
              >
                Read
              </option>
              <option
                value="none"
              >
                None
              </option>
            </select>
          </div>
        </div>
        <div
          className="book-title"
          data-teste="title"
        >
          To Kill a Mockingbird
        </div>
        <div
          className="book-authors"
          data-teste="authors"
        >
          Harper Lee
        </div>
        <div
          className="book-categories"
          data-teste="category"
        >
          COMPUTERS
        </div>
        <div
          className="book-rating"
        >
          <div
            className="dv-star-rating dv-star-rating-non-editable"
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_5"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={5}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_5"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_4"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={4}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_4"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_3"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={3}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_3"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_2"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={2}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_2"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_1"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={1}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_1"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
          </div>
        </div>
        <a
          className="book-more-info"
          data-teste="infoLink"
          href="https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api"
          onClick={[Function]}
          target="_blank"
          title="More info"
        >
          [ + ] More Info
        </a>
      </div>
    </li>
    <li>
      <div
        className="book"
      >
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            data-teste="imageLinks"
            style={
              Object {
                "backgroundImage": "url(./icons/placeholder.png)",
                "height": 192,
                "width": 128,
              }
            }
          />
          <div
            className="book-shelf-changer"
          >
            <select
              defaultValue={undefined}
              onChange={[Function]}
            >
              <option
                disabled={true}
              >
                Move to...
              </option>
              <option
                data-teste="shelf"
                value="currentlyReading"
              >
                Currently Reading
              </option>
              <option
                data-teste="shelf"
                value="wantToRead"
              >
                Want to Read
              </option>
              <option
                data-teste="shelf"
                value="read"
              >
                Read
              </option>
              <option
                value="none"
              >
                None
              </option>
            </select>
          </div>
        </div>
        <div
          className="book-title"
          data-teste="title"
        >
          To Kill a Mockingbird
        </div>
        <div
          className="book-authors"
          data-teste="authors"
        >
          Harper Lee
        </div>
        <div
          className="book-authors"
          data-teste="authors"
        >
          Teste2
        </div>
        <div
          className="book-rating"
        >
          <div
            className="dv-star-rating dv-star-rating-non-editable"
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_5"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={5}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_5"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_4"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={4}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_4"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_3"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={3}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_3"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_2"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={2}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_2"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_1"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={1}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_1"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
          </div>
        </div>
        <a
          className="book-more-info"
          data-teste="infoLink"
          href="https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api"
          onClick={[Function]}
          target="_blank"
          title="More info"
        >
          [ + ] More Info
        </a>
      </div>
    </li>
    <li>
      <div
        className="book"
      >
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            data-teste="imageLinks"
            style={
              Object {
                "backgroundImage": "url(http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api)",
                "height": 192,
                "width": 128,
              }
            }
          />
          <div
            className="book-shelf-changer"
          >
            <select
              defaultValue={undefined}
              onChange={[Function]}
            >
              <option
                disabled={true}
              >
                Move to...
              </option>
              <option
                data-teste="shelf"
                value="currentlyReading"
              >
                Currently Reading
              </option>
              <option
                data-teste="shelf"
                value="wantToRead"
              >
                Want to Read
              </option>
              <option
                data-teste="shelf"
                value="read"
              >
                Read
              </option>
              <option
                value="none"
              >
                None
              </option>
            </select>
          </div>
        </div>
        <div
          className="book-title"
          data-teste="title"
        >
          To Kill a Mockingbird
        </div>
        <div
          className="book-rating"
        >
          <div
            className="dv-star-rating dv-star-rating-non-editable"
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_5"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={5}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_5"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_4"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={4}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_4"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_3"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={3}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_3"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_2"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={2}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_2"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="rating_1"
              name="rating"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={1}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="rating_1"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "default",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
          </div>
        </div>
        <a
          className="book-more-info"
          data-teste="infoLink"
          href="https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api"
          onClick={[Function]}
          target="_blank"
          title="More info"
        >
          [ + ] More Info
        </a>
      </div>
    </li>
  </ol>
</div>
`;
